components/more_direct_channels/more_direct_channels.jsx

```diff
@@ -77,9 +78,13 @@ export default class MoreDirectChannels extends React.Component {
         UserStore.addInTeamChangeListener(this.onChange);
         UserStore.addStatusesChangeListener(this.onChange);
         if (this.listType === 'any') {
-            this.props.actions.getProfiles(0, USERS_PER_PAGE * 2, false);
+            this.props.actions.getProfiles(0, USERS_PER_PAGE * 2, false).then(() => {
+                this.toggleChannelLoadingState();
+            });
         } else {
-            this.props.actions.getProfilesInTeam(TeamStore.getCurrentId(), 0, USERS_PER_PAGE * 2);
+            this.props.actions.getProfilesInTeam(TeamStore.getCurrentId(), 0, USERS_PER_PAGE * 2).then(() => {
+                this.toggleChannelLoadingState();
```

Comment:
Why not just passed a value (true or false) instead of passing undefined? Maybe like 
```
this.setChannelLoadingState(!this.state.loadingChannels);
```